File: src/dataset/typescript/armstrong-numbers_Exercism-typescript-FilipeCerejo/openAI/ChatGPT4_armstrong-numbers_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isArmstrongNumber = isArmstrongNumber;
function isArmstrongNumber(number) {
    const strNumber = number.toString();
    const length = strNumber.length;
    let sum = 0;
    for (let i = 0; i < length; i++) {
        const digit = strNumber.charCodeAt(i) - 48; // Using charCodeAt for efficient digit conversion
        sum += Math.pow(digit, length);
        if (sum > number)
            return false; // Early exit if sum exceeds the number
    }
    return sum === number;
}

File: src/dataset/c/armstrong-numbers_exercism-Yusufislamkacar/openAI/ChatGPT4_armstrong_numbers_v2.c
#include "armstrong_numbers.h"
#include <math.h>
#include <stdbool.h>

bool is_armstrong_number(int x) {
    if (x < 0) return false;

    int original = x, total = 0, digits = 0;
    int temp = x;

    // Calculate the number of digits
    while (temp > 0) {
        digits++;
        temp /= 10;
    }

    temp = x;

    // Calculate the Armstrong sum
    while (temp > 0) {
        int digit = temp % 10;
        total += pow(digit, digits);
        temp /= 10;
    }

    return total == original;
}